modules/services/model/src/main/java/au/com/shiftyjelly/pocketcasts/models/db/dao/ChapterDao.kt

```diff
@@ -46,4 +52,34 @@ abstract class ChapterDao {
     protected abstract fun _observerChaptersForEpisode(episodeUuid: String): Flow<List<Chapter>>
 
     fun observerChaptersForEpisode(episodeUuid: String) = _observerChaptersForEpisode(episodeUuid).distinctUntilChanged()
+
+    @Transaction
+    open suspend fun selectChapter(episodeUuid: String, chapterIndex: Int, select: Boolean, modifiedAt: Date = Date()) {
+        val episode = findPodcastEpisode(episodeUuid) ?: findUserEpisode(episodeUuid) ?: return
+        val deselectedChapters = episode.deselectedChapters
+        episode.deselectedChapters = when {
+            select && chapterIndex in deselectedChapters -> ChapterIndices((deselectedChapters - chapterIndex).distinct())
+            !select && chapterIndex !in deselectedChapters -> ChapterIndices((deselectedChapters + chapterIndex).distinct())
+            else -> return
+        }
+        episode.deselectedChaptersModified = modifiedAt
+        update(episode)
+    }
```

Comment:
Sorry, I might be misunderstanding this. Why does this need to be in a transaction if it's only updating one episode? Would it be safer to update only the episodes `deselectedChapters` and `deselectedChaptersModified` columns, rather than every column?

Reply:
> Why does this need to be in a transaction if it's only updating one episode?

It's a series of operations. First read then write.

> Would it be safer to update only the episodes `deselectedChapters` and `deselectedChaptersModified` columns, rather than every column?

What risk do you see?

Reply:
> What risk do you see?
If the playback progress changed while this method was happening would it be possible to overwrite the playback progress to a different value?

Reply:
> If the playback progress changed while this method was happening would it be possible to overwrite the playback progress to a different value?

No, because it is in transaction.

Reply:
No need to change it. Thanks for explaining.